### ISSUE TITLE
inplace/bin/haddock expects threaded rts (HSrts_thr)

### ISSUE THREAD

**Issue by angerman** [snowleopard/shaking-up-ghc#98]
Or at least that's what I think `HSrts_thr` is, the threaded rts variant.
```
/------------------------------------------------------------\
| Run Ghc Stage2 (package = haddock)                         |
|     input: utils/haddock/stage2/build/Main.o (and 55 more) |
| => output: inplace/bin/haddock                             |
\------------------------------------------------------------/
ld: library not found for -lHSrts_thr
clang: error: linker command failed with exit code 1 (use -v to see invocation)
`clang' failed in phase `Linker'. (Exit code: 1)
shakeArgsWith   0.000s    0%                           
Function shake  0.020s    0%                           
Database read   0.532s    0%                           
With database   0.021s    0%                           
Running rules  60.154s   99%  =========================
Total          60.727s  100%                           
Error when running Shake build system:
* inplace/bin/haddock
user error (Development.Shake.cmd, system command failed
Command: inplace/bin/ghc-stage2 -hisuf hi -osuf o -hcsuf hc -static -hide-all-packages -no-user-package-db -package-id Cabal-1.23.0.0 -package-id array-0.5.1.0 -package-id base-4.9.0.0 -package-id bytestring-0.10.7.0 -package-id containers-0.5.7.1 -package-id deepseq-1.4.2.0 -package-id directory-1.2.5.0 -package-id filepath-1.4.1.0 -package-id ghc-8.1 -package-id ghc-boot-8.1 -package-id transformers-0.5.0.0 -package-id xhtml-3000.2.1 -i -iutils/haddock/stage2/build -iutils/haddock/stage2/build/autogen -iutils/haddock/driver -iutils/haddock/haddock-api/src -iutils/haddock/haddock-library/vendor/attoparsec-0.12.1.1 -iutils/haddock/haddock-library/src -Iutils/haddock/stage2/build -Iutils/haddock/stage2/build/autogen -I/Users/angerman/Projects/Haskell/ghc/compiler/. -I/Users/angerman/Projects/Haskell/ghc/compiler/parser -I/Users/angerman/Projects/Haskell/ghc/compiler/utils -I/Users/angerman/Projects/Haskell/ghc/compiler/../rts/dist/build -I/Users/angerman/Projects/Haskell/ghc/libraries/process/include -I/Users/angerman/Projects/Haskell/ghc/libraries/directory/include -I/Users/angerman/Projects/Haskell/ghc/libraries/unix/include -I/Users/angerman/Projects/Haskell/ghc/libraries/time/lib/include -I/Users/angerman/Projects/Haskell/ghc/libraries/containers/include -I/Users/angerman/Projects/Haskell/ghc/libraries/bytestring/include -I/Users/angerman/Projects/Haskell/ghc/libraries/base/include -I/Users/angerman/Projects/Haskell/ghc/libraries/integer-gmp/include -I/Users/angerman/Projects/haskell/ghc/rts/stage1/build -I/Users/angerman/Projects/haskell/ghc/includes -I/Users/angerman/Projects/haskell/ghc/includes/dist-derivedconstants/header -optP-include -optPutils/haddock/stage2/build/autogen/cabal_macros.h -funbox-strict-fields -Wall -fwarn-tabs -O2 -threaded -XHaskell2010 -optP-DIN_GHC_TREE -odir utils/haddock/stage2/build -hidir utils/haddock/stage2/build -stubdir utils/haddock/stage2/build -rtsopts -H32m -O2 -Wall -fwarn-tabs -no-auto-link-packages -optl-liconv -optl-lm -optl-ldl -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/xhtml/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/compiler/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/hpc/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/hoopl/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/ghci/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/transformers/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/template-haskell/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/ghc-boot/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/Cabal/Cabal/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/process/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/pretty/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/directory/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/unix/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/time/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/filepath/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/binary/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/containers/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/bytestring/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/deepseq/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/array/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/base/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/integer-gmp/stage1/build -optl-L/Users/angerman/Projects/Haskell/ghc/libraries/ghc-prim/stage1/build -optl-L/Users/angerman/Projects/haskell/ghc/rts/stage1/build utils/haddock/stage2/build/Main.o utils/haddock/stage2/build/Documentation/Haddock/Parser.o utils/haddock/stage2/build/Documentation/Haddock/Parser/Monad.o utils/haddock/stage2/build/Documentation/Haddock/Types.o utils/haddock/stage2/build/Documentation/Haddock/Doc.o utils/haddock/stage2/build/Data/Attoparsec.o utils/haddock/stage2/build/Data/Attoparsec/ByteString.o utils/haddock/stage2/build/Data/Attoparsec/ByteString/Buffer.o utils/haddock/stage2/build/Data/Attoparsec/ByteString/Char8.o utils/haddock/stage2/build/Data/Attoparsec/ByteString/FastSet.o utils/haddock/stage2/build/Data/Attoparsec/ByteString/Internal.o utils/haddock/stage2/build/Data/Attoparsec/Combinator.o utils/haddock/stage2/build/Data/Attoparsec/Internal.o utils/haddock/stage2/build/Data/Attoparsec/Internal/Fhthagn.o utils/haddock/stage2/build/Data/Attoparsec/Internal/Types.o utils/haddock/stage2/build/Data/Attoparsec/Number.o utils/haddock/stage2/build/Documentation/Haddock/Utf8.o utils/haddock/stage2/build/Documentation/Haddock/Parser/Util.o utils/haddock/stage2/build/Documentation/Haddock.o utils/haddock/stage2/build/Haddock.o utils/haddock/stage2/build/Haddock/Interface.o utils/haddock/stage2/build/Haddock/Interface/Rename.o utils/haddock/stage2/build/Haddock/Interface/Create.o utils/haddock/stage2/build/Haddock/Interface/AttachInstances.o utils/haddock/stage2/build/Haddock/Interface/LexParseRn.o utils/haddock/stage2/build/Haddock/Interface/ParseModuleHeader.o utils/haddock/stage2/build/Haddock/Interface/Specialize.o utils/haddock/stage2/build/Haddock/Parser.o utils/haddock/stage2/build/Haddock/Utils.o utils/haddock/stage2/build/Haddock/Backends/Xhtml.o utils/haddock/stage2/build/Haddock/Backends/Xhtml/Decl.o utils/haddock/stage2/build/Haddock/Backends/Xhtml/DocMarkup.o utils/haddock/stage2/build/Haddock/Backends/Xhtml/Layout.o utils/haddock/stage2/build/Haddock/Backends/Xhtml/Names.o utils/haddock/stage2/build/Haddock/Backends/Xhtml/Themes.o utils/haddock/stage2/build/Haddock/Backends/Xhtml/Types.o utils/haddock/stage2/build/Haddock/Backends/Xhtml/Utils.o utils/haddock/stage2/build/Haddock/Backends/LaTeX.o utils/haddock/stage2/build/Haddock/Backends/HaddockDB.o utils/haddock/stage2/build/Haddock/Backends/Hoogle.o utils/haddock/stage2/build/Haddock/Backends/Hyperlinker.o utils/haddock/stage2/build/Haddock/Backends/Hyperlinker/Ast.o utils/haddock/stage2/build/Haddock/Backends/Hyperlinker/Parser.o utils/haddock/stage2/build/Haddock/Backends/Hyperlinker/Renderer.o utils/haddock/stage2/build/Haddock/Backends/Hyperlinker/Types.o utils/haddock/stage2/build/Haddock/Backends/Hyperlinker/Utils.o utils/haddock/stage2/build/Haddock/ModuleTree.o utils/haddock/stage2/build/Haddock/Types.o utils/haddock/stage2/build/Haddock/Doc.o utils/haddock/stage2/build/Haddock/Version.o utils/haddock/stage2/build/Haddock/InterfaceFile.o utils/haddock/stage2/build/Haddock/Options.o utils/haddock/stage2/build/Haddock/GhcUtils.o utils/haddock/stage2/build/Haddock/Syb.o utils/haddock/stage2/build/Haddock/Convert.o utils/haddock/stage2/build/Paths_haddock.o -o inplace/bin/haddock
Exit code: 1
Stderr:
ld: library not found for -lHSrts_thr
clang: error: linker command failed with exit code 1 (use -v to see invocation)
`clang' failed in phase `Linker'. (Exit code: 1)
)
```

**Comment by angerman** [comment]
As `haddock` requires threaded rts, this can only be fixed, once #100 is done. In the meantime setting
```
buildHaddock = return False
```
in `Settings/User.hs`.

**Comment by snowleopard** [comment]
Now fixed. Haddock documentation is not built by default though, use `--haddock` flag to enable it.

Note: `threaded` way still needs a lot of work.
